brayns/io/XYZBLoader.cpp

```diff
@@ -98,25 +98,37 @@ ModelDescriptorPtr XYZBLoader::importFromBlob(
         updateProgress(msg.str(), i++, numlines);
     }
 
-    const float maxDim = model->getBounds().getSize().find_max();
-    if (maxDim < 100 * _geometryParameters.getRadiusMultiplier())
-    {
-        const float newRadius = maxDim / 100.f;
-        BRAYNS_WARN << "Given radius "
-                    << _geometryParameters.getRadiusMultiplier()
-                    << " is too big for this scene, using radius " << newRadius
-                    << " now" << std::endl;
-
-        for (i = 0; i < numlines; ++i)
-            spheres[i + startOffset].radius = newRadius;
-    }
+    // Find an appropriate mean radius to avoid overlaps of the spheres, see
+    // https://en.wikipedia.org/wiki/Wigner%E2%80%93Seitz_radius
+    const auto volume = bbox.getSize().product();
+    const float meanRadius =
+        std::pow((3.f / (4.f * M_PI * (numlines / volume))), 1.f / 3.f);
+
+    // resize the spheres to the new mean radius
+    for (i = 0; i < numlines; ++i)
+        spheres[i + startOffset].radius = meanRadius;
 
     Transformation transformation;
     transformation.setRotationCenter(model->getBounds().getCenter());
-
     auto modelDescriptor =
         std::make_shared<ModelDescriptor>(std::move(model), blob.name);
     modelDescriptor->setTransformation(transformation);
+
+    PropertyMap::Property radiusProperty("radius", "Point size", meanRadius,
+                                         {0, meanRadius * 2});
+    radiusProperty.setChangedCallback([
+        materialId, modelDesc = std::weak_ptr<ModelDescriptor>(modelDescriptor)
+    ](const auto& property) {
+        if (auto modelDesc_ = modelDesc.lock())
+        {
+            const auto newRadius = property.template get<float>();
+            for (auto& sphere : modelDesc_->getModel().getSpheres()[materialId])
+                sphere.radius = newRadius;
+        }
+    });
+    PropertyMap properties;
+    properties.setProperty(radiusProperty);
+    modelDescriptor->setProperties(properties);
```

Comment:
How difficult is to support a different radius for each sphere? If the input is 3D vectors you use this code path and if it's 4D you take the forth coordinate as the radius. That would be useful for many use cases.

Reply:
That's not feasible from the UI, and the radius is not part of the input file,  hence every sphere hast the same radius.

Reply:
In fact, I'm suggesting to extend the input file, which as far as I know is just a text.
The UI could expose a multiplicative factor, not the absolute radius to be used. That way you could still have explicit radius in the file. As a matter of fact, the multiplicative factor is something useful for animating groups of spheres (e.g. to show a selection when colors are already used for something else)